components/trace_metric/src/metric.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{fmt, time::Duration};
+
+#[derive(Clone)]
+pub struct MetricValue<T: Clone + fmt::Debug> {
+    pub name: String,
+    pub val: T,
+}
+
+#[derive(Clone)]
+pub enum Metric {
+    Boolean(MetricValue<bool>),
+    Counter(MetricValue<usize>),
+    Elapsed(MetricValue<Duration>),
```

Comment:
Timer?

Reply:
I guess `Timer` is not a good name here because `Timer` usually provides some utility methods e.g. start and stop rather than just a Duration.

Reply:
Boolean is the basic data type, seems a bit strange?

Reply:
Yes, usually it's called `Gauge`